app/lib/actions/pulp3/orchestration/orphan_cleanup/remove_orphans.rb

```diff
@@ -6,7 +6,10 @@ class RemoveOrphans < Pulp::Abstract
           def plan(proxy)
             if proxy.pulp3_enabled?
               sequence do
-                plan_action(Actions::Pulp3::OrphanCleanup::RemoveUnneededRepos, proxy) if proxy.pulp_mirror?
+                if proxy.pulp_mirror?
+                  plan_action(Actions::Pulp3::OrphanCleanup::RemoveUnneededRepos, proxy)
+                  plan_action(Actions::Pulp3::OrphanCleanup::DeleteOrphanRepositoryVersionsForMirror, proxy)
```

Comment:
I kinda like the idea of just having a 'DeleteOrphanRepositoryVersions' action, and it make the decision to do it for a mirror or a master (instead of having two 'dumber' actions), any thoughts about that?